documentation/design.md

```diff
@@ -0,0 +1,61 @@
+# Microsoft.Build.Prediction Design Notes
```

Comment:
Note that this is just a copy from the BuildXL repo with link fixups. I plan on iterating on this as we go.